### ISSUE TITLE
Pauses encoding when screen turned off [Asus Rog Laptop, Windows 11]

### ISSUE THREAD

**Issue by AircraftRu** [RandomEngy/VidCoder#1183]
### Problem Description

The converter pauses encoding when screen turned off,  when I log in it continues
Asus Rog Laptop, Windows 11 Pro
Laptop plugged in
Procces is set to Normal

### What version of VidCoder are you running?

8.26 Portable

**Comment by Neipas09** [comment]
Check your power settings.
Make sure that you aren't set to turn off your HDD when it does screen off.
I would also recommend changing your Power Setting to Ultimate Performance, and then see if it does the same thing.

**Comment by RandomEngy** [comment]
How is the screen turning off, here? VidCoder sets a flag to tell Windows not to automatically go into sleep mode while encoding. But there may be manually force sleep mode, like closing the laptop lid.

**Comment by AircraftRu** [comment]
> Check your power settings. Make sure that you aren't set to turn off your HDD when it does screen off. I would also recommend changing your Power Setting to Ultimate Performance, and then see if it does the same thing.

Yes, HDD is set to turn off after 1 minute after innactivity. Screen is set to 5 mins. The power settings is set to Perfomance mode

**Comment by AircraftRu** [comment]
> How is the screen turning off, here? VidCoder sets a flag to tell Windows not to automatically go into sleep mode while encoding. But there may be manually force sleep mode, like closing the laptop lid.

Screen is off after 5 mins. Of course I don't close lid, I just leave my laptop as is and don't even lock it. The laptop is set to go sleep after 5 hours.
I didn't notice VidCoder set a flag on that. My laptop is on, but VidCoder paused or slowed encoding, because the fans were quite. When I touch my laptop the fans start to work with pefomance loud and the progress continues after a long time when it would already be finished

**Comment by RandomEngy** [comment]
One other possibility is that the encode auto-paused from being low on disk space or battery.

VidCoder calls `SetThreadExecutionState(EXECUTION_STATE.ES_CONTINUOUS | EXECUTION_STATE.ES_SYSTEM_REQUIRED)` when starting the queue, and that flag is only cleared when the queue completes processing. You can open a command prompt in Admin mode and check with `powercfg /requests`

<img width="811" alt="image" src="https://github.com/RandomEngy/VidCoder/assets/9450861/de890026-51ac-499c-b876-c510530bb55a">

If VidCoder shows up in the System section, it's asking for the system to remain active. If your system goes to sleep even with that request active, there may be some OS-level customization that's overriding that.

**Comment by AircraftRu** [comment]
My disc has a lot of space and laptop is always plugged in.

![image](https://github.com/RandomEngy/VidCoder/assets/17082164/fa155620-08cb-4d6b-b154-bc24dfdb2d3d)

My laptop doesn't go to sleep, that's not the point, the app VidCoder 8.26 pauses processes while the system is atill active, just the screen is off.

**Comment by RandomEngy** [comment]
VidCoder does not have any code that pauses encoding when the screen is off. I can see from the powercfg output there that it is also asking the system not to go into sleep mode. Since you have enough disk space and power, the only thing I can think of is that some power-saving behavior in the OS, drivers, or hardware of your laptop is overriding VidCoder's system request and preventing it from using CPU when the screen is off.

**Comment by AircraftRu** [comment]
> VidCoder does not have any code that pauses encoding when the screen is off. I can see from the powercfg output there that it is also asking the system not to go into sleep mode. Since you have enough disk space and power, the only thing I can think of is that some power-saving behavior in the OS, drivers, or hardware of your laptop is overriding VidCoder's system request and preventing it from using CPU when the screen is off.

I don't know what it mignt be
